### ISSUE TITLE
not return error when parsing unescaped characters that MUST be escaped as JSON RFC

### ISSUE THREAD

**Issue by liuq19** [bytedance/sonic#95]
output: `should have unmarshal error like &json.SyntaxError{msg:"invalid character '\\b' in string literal", Offset:2}`

```
func TestInvalideUnicode(t *testing.T) {
	var sonicv, stdv interface{}
	data := []byte("\"\b\"")
	sonicerr := Unmarshal(data, &sonicv)
	stderr := json.Unmarshal(data, &stdv)
	if stderr != nil {
		if sonicerr == nil {
			t.Errorf("should have unmarshal error like %#v\n", stderr)
		}
		return
	}
	if !reflect.DeepEqual(sonicv, stdv) {
		t.Errorf("Unmarshal %#v\n want obj:\n %#v\n get obj:\n %#v\n", data, stdv, sonicv)
	}
}
```

**Comment by liuq19** [comment]
The problem will be solved at #96

**Comment by liuq19** [comment]
You can validate and then unmarshal if needed.  #189
